ui/main.py

```diff
@@ -192,6 +200,14 @@ def change_files(config, state):
     if state['backing_up_log'] == 'start':
         store.dispatch(actions.backup_log('in progress'))
         backup_log(config)
+    if state['update_ui'] == 'start':
+        log.info("update ui = start")
+        if utility.find_ui_update(config):
+            store.dispatch(actions.update_ui('in progress'))
+            store.dispatch(actions.halt_ui(True))
```

Comment:
I think it would be clearer if the state consumers decides to halt if the update is in progress. I.e. you seem to have two bits of the state expressing the same thing here.